rkt/fetch.go

```diff
@@ -47,9 +59,106 @@ func init() {
 	cmdFetch.Flags().Var((*appAsc)(&rktApps), "signature", "local signature file to use in validating the preceding image")
 	cmdFetch.Flags().BoolVar(&flagStoreOnly, "store-only", false, "use only available images in the store (do not discover or download from remote URLs)")
 	cmdFetch.Flags().BoolVar(&flagNoStore, "no-store", false, "fetch images ignoring the local store")
+	cmdFetch.Flags().IntVar(&flagP2pDuration, "p2p-duration", 10, "p2p continue service duration (minutes) after download finished")
+}
+
+func p2pFetch(args []string) int {
+	if len(args) < 1 {
+		fmt.Printf("rkt fetch must provide a tottent file.")
+		return 1
+	}
+
+	duration := strconv.Itoa(flagP2pDuration)
+	//start p2p client
+	var cmd *exec.Cmd
+	cmd = exec.Command("./torrent", args[0], duration)
+	if err := cmd.Start(); err != nil {
+		fmt.Printf("start p2p process err:%s\n", err)
+		return 1
+	}
+
+	pid := cmd.Process.Pid
+
+	//handle Ctrl+C signal
+	go func() {
+		signalChan := make(chan os.Signal, 1)
+		signal.Notify(signalChan, os.Interrupt, os.Kill)
+
+		<-signalChan
+		if err := syscall.Kill(pid, 9); err != nil {
+			fmt.Printf("kill p2p client process err:%s\n", err)
```

Comment:
Error messages should go to stderr, using statement like in line 163. If You want to output some informational things like in 127 line - You should use statement like in line 201.